hippod/report_generator.py

```diff
@@ -55,79 +55,74 @@ def get_format_snippet(app, data_id):
             return snippet_format
 
 
+        def get_dst_path(self, data_type, sub_dir, name):
+            if data_type == '.png':
+                dst_path = os.path.join(sub_dir, '{}.png'.format(name))
+            elif data_type == '.jpg':
+                dst_path = os.path.join(sub_dir, '{}.jpg'.format(name))
+            elif data_type == '.jpeg':
+                dst_path = os.path.join(sub_dir, '{}.jpeg'.format(name))
+            elif data_type == '.gif':
+                dst_path = os.path.join(sub_dir, '{}.gif'.format(name))
+            else:
+                # FIXME: not sure, but this function should return. If
+                # not dst_path is undefined and will definitly crash some
+                # lines later!
+                log.error("data type not supported: {}".format(data_type))
+                return None
+            return dst_path
+
+
+        def copy_to_dst(self, src_path, dst_path):
+            with open(src_path, 'rb') as file:
+                data = file.read()
+                #data = zlib.decompress(data)
+                data += b'==='                                              # arrange that correctly!
```

Comment:
Actually thats not new code. Previously there were some problems decoding the data.